tests/director/gateway-integration/automatic_status_update_test.go

```diff
@@ -0,0 +1,182 @@
+		defer unregisterIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+
+		t.Log("Request Client Credentials for Integration System")
+		intSysOauthCredentialData := requestClientCredentialsForIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+
+		t.Log("Issue a Hydra token with Client Credentials")
+		accessToken := getAccessToken(t, intSysOauthCredentialData, "application:write")
+		oauthGraphQLClient := gql.NewAuthorizedGraphQLClientWithCustomURL(accessToken, fmt.Sprintf("https://compass-gateway-auth-oauth.%s/director/graphql", domain))
+
+		t.Log("Register application as Integration System")
+		appInput := graphql.ApplicationRegisterInput{
+			Name:                "app-is",
+			ProviderName:        ptr.String("compass"),
+			IntegrationSystemID: &intSys.ID,
+		}
+		app, err := registerApplicationWithinTenant(t, ctx, oauthGraphQLClient, tenant, appInput)
+		require.NoError(t, err)
+
+		defer unregisterApplication(t, ctx, oauthGraphQLClient, tenant, app.ID)
+
+		assert.Equal(t, graphql.ApplicationStatusConditionInitial, app.Status.Condition)
+
+		t.Log("Update the application")
+		status := graphql.ApplicationStatusConditionFailed
+		appUpdateInput := graphql.ApplicationUpdateInput{
+			Description:     str.Ptr("New description"),
+			StatusCondition: &status,
+		}
+		appUpdated, err := updateApplicationWithinTenant(t, ctx, oauthGraphQLClient, tenant, app.ID, appUpdateInput)
+		require.NoError(t, err)
+
+		t.Log("Ensure the status condition")
+		assert.Equal(t, graphql.ApplicationStatusConditionFailed, appUpdated.Status.Condition)
+	})
+
+	t.Run("Test automatic status update as Application", func(t *testing.T) {
+		appInput := graphql.ApplicationRegisterInput{
+			Name: "test-app",
+			Labels: &graphql.Labels{
+				"scenarios": []interface{}{"DEFAULT"},
+			},
+		}
+
+		t.Log("Register Application with Dex id token")
+		app := registerApplicationFromInputWithinTenant(t, ctx, dexGraphQLClient, tenant, appInput)
+
+		t.Logf("Registered Application with [id=%s]", app.ID)
+		defer unregisterApplication(t, ctx, dexGraphQLClient, tenant, app.ID)
+
+		t.Log("Request Client Credentials for Application")
+		appAuth := requestClientCredentialsForApplication(t, ctx, dexGraphQLClient, tenant, app.ID)
+		appOauthCredentialData, ok := appAuth.Auth.Credential.(*graphql.OAuthCredentialData)
+		require.True(t, ok)
+		require.NotEmpty(t, appOauthCredentialData.ClientSecret)
+		require.NotEmpty(t, appOauthCredentialData.ClientID)
+
+		t.Log("Issue a Hydra token with Client Credentials")
+		accessToken := getAccessToken(t, appOauthCredentialData, "application:read")
+		oauthGraphQLClient := gql.NewAuthorizedGraphQLClientWithCustomURL(accessToken, fmt.Sprintf("https://compass-gateway-auth-oauth.%s/director/graphql", domain))
+
+		t.Log("Requesting Viewer as Application")
+		viewer := graphql.Viewer{}
+		req := fixGetViewerRequest()
+
+		assert.Equal(t, graphql.ApplicationStatusConditionInitial, app.Status.Condition)
```

Comment:
Why do you check here the status?
I think this assert is in the middle of nowhere.

The same situation is for runtime test.